elastic/kibana
# [Advanced Settings] Rename defaultIndex setting to defaultDataView

Closes https://github.com/elastic/kibana/issues/168541

## Summary

This PR renames the ID of the `defaultIndex` setting to `defaultDataView`.

**How to test:**
1. Start Es with `yarn es snaphsot`/`yarn es serverless` and Kibana with `yarn start`/`yarn serverless-{es/oblt/security}`.
2. Go to Management -> Advanced Settings
3. Verify that the setting is now called `defaultDataView`.
